internal/pkg/log/logger.go

```diff
@@ -43,17 +43,17 @@ func (l *zapLogger) WithComponent(component string) Logger {
 	if l.component != "" {
 		component = l.component + "." + component
 	}
-	clone := loggerFromZapCore(l.core)
+	clone := *l
 	clone.component = component
-	return clone
+	return &clone
 }
```

Comment:
There is not necessary to create a new wrapped logger instance (`zap.New(core).Sugar()`)  in `WithComponent`.

Clone of the `zapLogger` structure is enough.